file_optical.c

```diff
@@ -1670,13 +1670,68 @@ static int fbo_handle_cmd(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 	return 0;
 }
 
+static int fbo_update_cfgstring(struct tcmu_device *dev,
+				struct tcmulib_cfg_info *cfg)
+{
+	char *oldptr;
+	int ret;
+
+	if (!tcmu_check_state(dev, TCMU_DEV_STATE_CAPACITY_CHANGE_FOR_CONFIG_CHANGE)) {
```

Comment:
I did not get why you need to change the size and cfg string?

Is it just to enforce that if the size did change it would always need to come before the cfg string part part of the update?